CppCustomVisualizer2/dll/RootVisualizer.cpp

```diff
@@ -385,7 +388,7 @@ HRESULT CRootVisualizer::GetSize(
 
     LPCWSTR sizeStr = pValue->Value();
     LPWSTR endPtr;
-    *pSize = wcstoull(sizeStr, &endPtr, 0);
+    *pSize = wcstoul(sizeStr, &endPtr, 0);
```

Comment:
I see why I think (GetSize might get a strSize that is higher than 32 bit max back from DkmEvaluationResult).

What should we do at passing it into `DkmEvaluationResultEnumContext::Create`?

I think maybe `(DWORD) min(m_size, (unsigned long long)UINT_MAX)` ? That way if it's larger than unsigned long long, it gets truncated down to UINT_MAX?

Reply:
I'm not familiar with DkmEvaluationResultEnumContext, but I imagine having billions of items is uncommon/impossible?

Reply:
Ah, now I see why you did it that way you had it in iteration 1. 

To explain, this class is reading the size field of an STL collection class, which uses `size_t` for size, and therefore this code used `wcstoull` to convert that size (as `size_t` is 64-bit if the target process is 64-bit, and therefore it always wanted to use a 64-bit size).

That said, since this class is ultimately going to send back results to the EE using a count that maxes out at MAXDWORD your fix was okay. But, I think I marginally like your new fix better since it makes this more apparent that we are truncating to MAXDWORD. But either way. If you do prefer your iteration 1 fix, we probably should -- 
1. Double check that `wcstoul` will saturate at MAXDWORD (if there is a larger number). I believe it will.
2. Add a comment.

Reply:
I prefer my second fix since it's less intrusive and already done. Thanks for explaining!